File: web2-Project/src/app/entities/flight/flight.ts
import { Presedanje } from '../flight-presedanje/presedanje';
import { Aeroplane } from '../aeroplane/aeroplane';
import { AvioLuggage } from '../avio-luggage/avio-luggage';
import { FlightReservation } from '../flight-reservation/flight-reservation';
import { Address } from '../address/address';

export class Flight {
    id: number;
    company: string;
    idCompany: number;
    logo: string;
    from: Address; //
    to: Address; //
    destImg: string; //
    datumPolaska: Date; //
    datumSletanja: Date;//
    prise: number; //
    priceTwoWay: number; //* cena povratne karte
    vremePutovanja: string;
    duzinaPutovanja: number;
    presedanje: Presedanje;
    aeroplane: Aeroplane;
    luggage: AvioLuggage;
    
    // polja koja ne prosledjujem kroz konstruktor, jer nisu obavezna za let
    numOfFastReseravtions: number; // broj sedista koja moze da ima za brzu rezervaciju
    discountForFastReservation: number; // popust za brzu rezervaciju
    ocene: Array<number>;

    //allReservations: Array<FlightReservation>;
    
    constructor(id: number, company: string, idCompany: number, logo: string, from: Address, to: Address, destImg: string,
                datumPolaska: Date, datumSletanja: Date, prise: number, priceTwoWay: number,
                vremePutovanja: string, duzinaPutovanja: number, presedanje: Presedanje, aeroplane: Aeroplane,
                luggage: AvioLuggage) {
        this.id = id;
        this.company = company;
        this.idCompany = idCompany;
        this.logo = logo;
        this.from = from;
        this.to = to;
        this.destImg = destImg;
        this.datumPolaska = datumPolaska;
        this.datumSletanja = datumSletanja;
        this.prise = prise;
        this.priceTwoWay = priceTwoWay;
        this.vremePutovanja = vremePutovanja;
        this.duzinaPutovanja = duzinaPutovanja;
        this.presedanje = presedanje;
        this.aeroplane = aeroplane;
        this.ocene = new Array<number>();
        this.luggage = luggage;
        //this.allReservations = new Array<FlightReservation>();
    }
}

File: MAANPP20/MAANPP20/MAANPP20/Models/Flights/Flight.cs
using MAANPP20.Models.Common;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace MAANPP20.Models.Flights
{
    public class Flight
    {
        [Key]
        public int id { get; set; }

        public string company { get; set; }

        public int idCompany { get; set; }

        public string logo { get; set; }

        [ForeignKey("from")]
        public int addressFromId { get; set; }
        public Address from { get; set; }

        [ForeignKey("to")]
        public int addressToId { get; set; }
        public Address to { get; set; }

        public string destImg { get; set; }

        public DateTime datumPolaska { get; set; }

        public DateTime datumSletanja { get; set; }

        public double prise { get; set; }

        public double priceTwoWay { get; set; }

        public string vremePutovanja { get; set; }

        public double duzinaPutovanja { get; set; }

        [ForeignKey("presedanje")]
        public int presedanjeId { get; set; }
        public Presedanje presedanje { get; set; }

        [ForeignKey("aeroplane")]
        public int aeroplaneId { get; set; }
        public Aeroplane aeroplane { get; set; }

        [ForeignKey("luggage")]
        public int luggageId { get; set; }
        public AvioLuggage luggage { get; set; }

        public int numOfFastReseravtions { get; set; }

        public double discountForFastReservation { get; set; }

        public ICollection<DoubleForICollection> ocene { get; set; }

        public ICollection<AvioSediste> allSeatsForThisFlight { get; set; }

        public bool deleted { get; set; }
    }
}
